stub/src/main/java/io/grpc/stub/ClientCalls.java

```diff
@@ -708,10 +708,10 @@ void onStart() {
   }
 
   @SuppressWarnings("serial")
-  private static final class ThreadlessExecutor extends ConcurrentLinkedQueue<Runnable>
```

Comment:
This was done on purpose to reduce allocations. https://github.com/grpc/grpc-java/commit/5f88bc42bfeba902d92737e2a6a4c0dcbbff302e The ErrorProne complaint is sorta fair, but we aren't actually overriding any of the behavior, so it is reasonably safe for us to extend.

The real problem is Java encourages extension instead of composition.

Reply:
Ah, I was wondering why that had been done. I reverted that change.